027-network-policy-generation-feature-gate.md

```diff
@@ -0,0 +1,63 @@
+# Network Policy Generation Feature Gate
+
+The feature would consist of a new Feature Gate which can be used to disable the generation of Network Policies by Strimzi, where the user wishes to set up a more restrictive Network Policy set.
+
+## Current situation
+
+Currently, Strimzi generates Network Policies which allow all pods in the deployment to connect to Kafka, Zookeeper and any other service that it has generated.
+
+There is no way to stop this generation.
+
+It is possible, in the select case of listeners, to further restrict the Ingress using NetworkPolicyPeers, however this does not extend to all Network Policy generation, for example Metrics.
+It is also not possible to set up Egress Network Policies. 
+
+## Motivation
+
+The current Network Policies that are generated are great for getting Strimzi up and running in a deployment where a deny-all network policy is in place.
+
+However, this does not allow users of Strimzi to reach a fully secured Network Policy set, due to Strimzi not providing the full feature set of Kubernetes Network Policies.
+
+By adding a feature gate, it allows users to disable this generation, and instead write fully secure Network Policies.
+
+This setup is valid in 2 situations:
+
+* A deployment with no deny-all Network Policy, where all pods can already communicate with each other.
+* A deployment where the consumer wishes to write their own secure Network Policies.
+
+## Proposal
+
+The proposal would be to introduce a Feature Gate which would provide the ability to turn of the generation of Network Policies by Strimzi.
+
+As the functionality already exists to generate Network Policies, a sensible naming of this gate would be `NetworkPolicyGeneration` which by default is set to `true`.
+If users require secure Network Policies, they can disable this using the regular Feature Gate syntax.
+This Feature Gate will be a long lived Feature Gate, and is not meant to phase in a feature. Instead, it should remain so users can turn off this functionality when needed.
```

Comment:
I think it would be maybe good to add some statement about alpha, beta, GA stages. I agree that this should be long-lived gate which should remain in the same default state. But maybe we want to still go through the alpha and beta stages in case we find out that it causes some unexpected problems and decide we want to implement this differently?

Reply:
Added a comment about it moving through the stages, however remaining the same value unlike other Feature Gates.